### ISSUE TITLE
Have an option to allow users to have as many tabs as needed (without using tabs group)

### ISSUE THREAD

**Issue by unsystemizer** [brave/browser-laptop#4986]
**Did you search for similar issues before submitting this one?**
Yes

**Describe the issue you encountered:**
When the number of tabs is more than the width of the browser window, it becomes very hard to navigate with the mouse between tabs that require "horizontal scrolling". 

**Expected behavior:**
I would expect something similar to what other good browsers do:
1.) do not create tab groups by default. I'm fine with 50 tabs and don't want to horizontally scroll among them. Maybe have "Enable grouping" in {10/20/x} tabs" for those who like this way, but this isn't standard in other browsers and shouldn't be enabled by default (but I'm fine with that as long as I could disable this behavior)
2.) provide incremental (smooth) horizontal scrolling (not in "screen width" increments) between tab "groups"
3.) provide the ability to pack more tabs in one screen (if I have 45 tabs on my large monitor, I don't need or want to scroll horizontally between many tab sets)
- Platform (Win7, 8, 10? macOS? Linux distro?):
  Win 10
- Brave Version:
  0.12.5
- Steps to reproduce:
  1. Open 20+ tabs in one Brave window (enough so that they spill over to require horizontal scrolling)
  2. Try to switch between the first and the last tab. Normally (with few tabs) it takes 1 click (on the target tab). Here it takes 100% more (click on the left or right horizontal scroll arrow to visit the next batch of tabs) if one uses the little tab navigation thing:
     ![image](https://cloud.githubusercontent.com/assets/13134193/19550506/59dd6814-96dc-11e6-873e-c0392d54bd3e.png)
     Without it (I'm not used to it because those bars are too tiny, and I'd rather have them disappear than be enlarged) it may take 3 clicks instead of 1 (if you have 3 "window widths" worth of tabs and need to navigate between window 1 and 3, you need to click several times on the navigation arrows).
  3. Additionally, because of this horizontal scrolling, it's hard to remember where the previous tab was. For example I go from "tab set in screen width number 3" (not sure what's the right term for that) to "number 1" and read something for a while. Then when I go back I need to find the correct place to click (horizontal tab set number 3) and only then can I see the tab. 
  4. A major sub-issue is when two tabs of interest are adjacent. Although they are next to each other, they may be in different tab sets, so just to switch between these two I need 2 clicks every time (plus the not-so-enjoyable "flashing" effect of the "tab preview" feature which hurts my eyes when I read in dark; consider that to go from tab 2 in group 2 to tab 8 in group 1, one has to click on the left hand "tab navigation arrow", which makes your mouse pointer pointed at the leftmost tab in group 1 and then you need to drag it all the way to the right to see tab 8, during which time all tabs are flashing (previewing) if you happen to drag your pointer across the tops of tabs 1-7; the same problem happens when you want to return tab 1 of tab group 2!).
- Screenshot if needed:
- Any related issues:
  https://github.com/brave/browser-laptop/issues/2334 - this isn't directly related but it would be nice if some "tab scrolling" improvements could be made in one go.

**Comment by bsclifton** [comment]
This is good feedback :smile: cc: @bradleyrichter

**Comment by bradleyrichter** [comment]
@unsystemizer Thanks for the detailed feedback. We will get there.  :  )

During initial research to improve tab overflow, we found that people either prefer Firefox or Chrome but most people disliked aspects of each, and wished for something better. 

We found that users open a new window (in both browsers) and start more tabs. Then they have a worse problem trying to locate and keep track of the window-separated tabs because they are disconnected. 

Chrome turns tabs into unrecognizable nubs fairly quickly. This forces some users to start pruning in a good way. Firefox creates difficulty in locating the tabs when you can only view a subset at any given time. 

Brave Tab sets tries to improve this main problem and ads better visual location. 

Soon, we will be adding horizontal scrolling like Firefox to what we have now. So basically, the tab set indicators will be "scroll jumpers" instead of separated sets. Combined with adjustable number of tabs per tab-set, later up to 100, you can potentially have the best of all 3 worlds. 

In the future, we plan to add organizational features to tab sets.

**Comment by bradleyrichter** [comment]
BTW - this is now partially solved. It no longer happens every time.

**Comment by cezaraugusto** [comment]
I think there's a great feedback here, and will change the title to something more general so we could have feedback from other users as well.

**Comment by cezaraugusto** [comment]
As a future reference for someone getting here:

tl;dr: in a future, as @bradleyrichter suggested, we would have horizontal scrolling like Firefox, and in a nutshell, tab set indicators will be "scroll jumpers" instead of separated sets. (more info on [his comments](https://github.com/brave/browser-laptop/issues/4986#issuecomment-255252021))

**Comment by bsclifton** [comment]
Closing as a duplicate of https://github.com/brave/browser-laptop/issues/879. I made sure to give a +1 and copied feedback 😄
